quill-core/src/main/scala/io/getquill/dsl/EncodingDsl.scala

```diff
@@ -1,15 +1,17 @@
 package io.getquill.dsl
 
-import scala.language.higherKinds
-import scala.annotation.compileTimeOnly
 import io.getquill.quotation.NonQuotedException
+
+import scala.annotation.compileTimeOnly
 import scala.language.experimental.macros
+import scala.language.higherKinds
 
 trait LowPriorityImplicits {
   this: EncodingDsl =>
 
-  implicit def materializeEncoder[T]: Encoder[T] = macro EncodingDslMacro.materializeEncoder[T]
-  implicit def materializeDecoder[T]: Decoder[T] = macro EncodingDslMacro.materializeDecoder[T]
+  implicit def materializeEncoder[T <: AnyVal]: Encoder[T] = macro EncodingDslMacro.materializeEncoder[T]
```

Comment:
Could you clarify why you needed to add the `<: AnyVal` constraint? The materializers also work for `AnyRef`s